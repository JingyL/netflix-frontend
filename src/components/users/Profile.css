.profile{
  height: 140vh;
}
.profile-title{
  color:white;
  font-family: 'Libre Baskerville', serif;
  font-size: 30px;
}

.profile-card{
  margin-top: 15vh;
  width: 40%;
  background-color: rgba(48, 1, 1, 0);
  position: absolute;
  transform: translate(-50%,-50%);
  top: 50%;
  left: 50%;
  border-radius: 10px;
  backdrop-filter: blur(10px);
  border: 2px solid rgba(255,255,255,0.1);
  box-shadow: 0 0 40px rgba(8,7,16,0.6);
  padding: 50px 35px;
}

.profile-label-font{
  color:white;
  font-family: 'Libre Baskerville', serif;
}

label{
  display: block;
  margin-top: 30px;
  font-size: 16px;
  font-weight: 500;
}

input{
  display: block;
  height: 4.5vh;
  width: 100%;
  background-color: rgba(255,255,255,0.07);
  margin-top: 5%;
  font-size: 14px;
  font-weight: 300;
  color:white;
}

.profile-page-btn{
  margin-top: 10%;
  width: 100%;
  background-color: #ffffff;
  color: #080710;
  padding: 15px 0;
  font-size: 18px;
  font-weight: 600;
  border-radius: 5px;
  cursor: pointer;
  margin-bottom: 30px;
}

.profile-username{
  text-align: left;
}

.successMsg{
  color:green;
}