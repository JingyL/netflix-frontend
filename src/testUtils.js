import React from "react";
import UserContext from "./hooks/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
};

const UserProvider =
    ({ children, currentUser = demoUser, addedMovies=[] }) => (
    <UserContext.Provider value={{ currentUser, addedMovies }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
