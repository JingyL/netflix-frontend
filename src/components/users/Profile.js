import React, { useContext, useState } from "react";
import "./Profile.css";
import UserContext from "../../hooks/UserContext";
import Navbar from "../navbar/Navbar";

// Profile Page which user can change their password and other info
function Profile({changeProfile}) {
  const { currentUser, setCurrentUser } = useContext(UserContext);
  const [successMsg, setSuccessMsg] = useState("");
  const [errorMsg, setErrorMsg] = useState("");
  const [formData, setFormData] = useState({
    firstName: currentUser.firstName,
    lastName: currentUser.lastName,
    password: "",
    email:currentUser.email
  });

  function handleChange(e) {
    e.persist();
    console.log(e.target.value)
    setFormData(f => ({ ...f, [e.target.name]: e.target.value }));
  }

  async function handleSubmit(e) {
    e.preventDefault();
    let res = await changeProfile(currentUser.username, formData);
    if (res["success"]){
      setSuccessMsg(res["success"]);
    }else{
      setSuccessMsg(res["error"]);
    }
  }


  return (
    <div className="profile">
      <Navbar></Navbar>
          <div className="profile-card">
          <h3 className="profile-title">Profile</h3>
            <form>
              <div className="profile-label-font">
                <label>Username</label>
                <p className="profile-username">{currentUser.username}</p>
              </div>

              <div className="profile-label-font">
                <label>First Name</label>
                <input
                  name="firstName"
                  value={currentUser.firstName}
                  onChange={handleChange}
                />
              </div>

              <div className="profile-label-font">
                <label>Last Name</label>
                <input
                  name="lastName"
                  value={currentUser.lastName}
                  onChange={handleChange}
                />
              </div>

              <div className="profile-label-font">
                <label>Email</label>
                <input
                  name="email"
                  value={currentUser.email}
                  onChange={handleChange}
                />
              </div>

              <div className="profile-label-font">
                <label>Confirm password to make changes:</label>
                <input
                  type="password"
                  name="password"
                  value={currentUser.password}
                  onChange={handleChange}
                />
              </div>

              {setSuccessMsg
              ? <p className="successMsg">{successMsg}</p>
              : <p className="errorMsg">{errorMsg}</p>}

              <button
                className="profile-page-btn"
                onClick={handleSubmit}
              >
                Save Changes
              </button>
            </form>
            </div>




    </div>

  );
}

export default Profile;