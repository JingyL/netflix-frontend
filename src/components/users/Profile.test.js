import React from "react";
import { render } from "@testing-library/react";
import Profile from "./Profile";
import { UserProvider } from "../../testUtils";
import { BrowserRouter } from "react-router-dom";


it("matches snapshot", function () {
  const { asFragment } = render(
    <BrowserRouter>
      <UserProvider>
        <Profile />
      </UserProvider>
    </BrowserRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
