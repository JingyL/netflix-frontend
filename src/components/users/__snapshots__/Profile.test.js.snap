// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="profile"
  >
    <div
      class="nav"
    >
      <a
        class="nav-logo"
        href="/movies"
      >
        <img
          alt="Netflix Logo"
          class="nav-logo"
          src="https://upload.wikimedia.org/wikipedia/commons/0/08/Netflix_2015_logo.svg"
        />
      </a>
      <a
        href="/search"
        style="text-decoration: none;"
      >
        <h3
          class="nav-text"
        >
          Search
        </h3>
      </a>
      <a
        href="/mylist"
        style="text-decoration: none;"
      >
        <h3
          class="nav-text"
        >
          My List
        </h3>
      </a>
      <a
        href="/profile"
        style="text-decoration: none;"
      >
        <h3
          class="nav-text"
        >
          Profile
        </h3>
      </a>
      <a
        class="nav-text-signout"
        href="/signin"
        style="text-decoration: none;"
      >
        <h3>
          Sign Out
        </h3>
      </a>
    </div>
    <div
      class="profile-card"
    >
      <h3
        class="profile-title"
      >
        Profile
      </h3>
      <form>
        <div
          class="profile-label-font"
        >
          <label>
            Username
          </label>
          <p
            class="profile-username"
          >
            testuser
          </p>
        </div>
        <div
          class="profile-label-font"
        >
          <label>
            First Name
          </label>
          <input
            name="firstName"
            value=""
          />
        </div>
        <div
          class="profile-label-font"
        >
          <label>
            Last Name
          </label>
          <input
            name="lastName"
            value=""
          />
        </div>
        <div
          class="profile-label-font"
        >
          <label>
            Email
          </label>
          <input
            name="email"
            placeholder="test@test.net"
            value="test@test.net"
          />
        </div>
        <div
          class="profile-label-font"
        >
          <label>
            Confirm password to make changes:
          </label>
          <input
            name="password"
            type="password"
            value=""
          />
        </div>
        <p
          class="successMsg"
        />
        <button
          class="profile-page-btn"
        >
          Save Changes
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
